pinecone/control/pinecone.py

```diff
@@ -1,22 +1,32 @@
 import time
 import warnings
 import logging
-from typing import Optional, Dict, Any, Union, List, cast, NamedTuple
+from typing import Optional, Dict, Any, Union, List, Tuple, cast, NamedTuple
```

Comment:
A decent amount of change was needed in this file due to changes in generated code related to index `spec`.